### ISSUE TITLE
Fluent setter for generated PHP-Objects

### ISSUE THREAD

**Issue by ElectroLutz** [google/protobuf#2944]
Hey,

I'm currently testing the PHP implementation of protobuf and I'm curious why the setters arent fluent.

Maybe it's that way for a reason but I currently don't see why.

If it's just not there, the following should do it (I never programed c++; so just a guess):

after https://github.com/google/protobuf/blob/master/src/google/protobuf/compiler/php/php_generator.cc#L516

add something like:
`printer->Print("return $this;\n");`

**Comment by TeBoring** [comment]
Protobuf will do type conversion if given type is not the same as the targeted. e.g. for setInt32($num_string), we will convert $num_string to integer. What do you expect to return in such case?

**Comment by TeBoring** [comment]
Also, feel free to add a proposal in https://docs.google.com/document/d/1fUi0lKpfR7gauaJdhjid1SStdLePREVeUnwfamrPRVk/edit#heading=h.v03byv1vu6dr

**Comment by ElectroLutz** [comment]
I think you misunderstood me. My issue is not about type convertion.

I wanted to ask why there is no 
`return $this;` 
at the end of each setSomething() method. So I can do things like 
`$foo->setX($string)->setY($number)->setZ($array);`

Just a fluent interface for the setters. This would save us some typing and should be a big deal.

Don't hesitate to ask me, when there are any open questions.

**Comment by ElectroLutz** [comment]
Any news on this one?

**Comment by ElectroLutz** [comment]
@TeBoring @xfxyjwf ??

**Comment by TeBoring** [comment]
Currently, I tend to leave this open. Waiting to see if there are a lot of people want this feature.

**Comment by michaelbausor** [comment]
+1, I think this would be a good feature to include. cc @bshaffer @tmatsuo

**Comment by bshaffer** [comment]
This would be a nice feature, given all the boilerplate setting of properties that's necessary for some Protobuf objects, i.e.

```php
$timeSeries = (new TimeSeries())
    ->setMetric($m)
    ->setResource($r)
    ->setPoints($points);
```

I agree this is easier to read as a single block than the current implementation:

```php
$timeSeries = new TimeSeries();
$timeSeries->setMetric($m);
$timeSeries->setResource($r);
$timeSeries->setPoints($points);
```

But I do not see this as a must-have. Other features like #3118 and #3119 are more important.

**Comment by mlosev** [comment]
+1 for fluent setters

**Comment by voksiv** [comment]
+1 fluent setter... We want this! We crave it! We pray for it!

**Comment by TeBoring** [comment]
I can add it later.

**Comment by ev4ern** [comment]
+1 for fluent setters

**Comment by Yakoot** [comment]
+1 for fluent setters

**Comment by GRbit** [comment]
It will be a handy feature, want it)

**Comment by Grey2k** [comment]
+1 for fluent setters

**Comment by bshaffer** [comment]
Wow they really like this feature in St Petersburg...

**Comment by waprin** [comment]
+1 for fluent setters

**Comment by bshaffer** [comment]
Resolved in #3130
